app/request_handlers.py

```diff
@@ -638,14 +707,10 @@ async def post(self, request):
             fulfilment_code_array = []
 
             try:
-                available_fulfilments = await RHSvc.get_fulfilment(
-                    request, region, 'SMS', 'UAC', fulfilment_individual)
-                if len(available_fulfilments) > 1:
-                    for fulfilment in available_fulfilments:
-                        if fulfilment['language'] == fulfilment_language:
-                            fulfilment_code_array.append(fulfilment['fulfilmentCode'])
-                else:
-                    fulfilment_code_array.append(available_fulfilments[0]['fulfilmentCode'])
+                available_fulfilments = \
+                    await RHSvc.survey_fulfilments_by_type(request, 'SMS', survey_id, fulfilment_language)
+
+                fulfilment_code_array.append(available_fulfilments)
 
                 try:
                     await RHSvc.request_fulfilment_sms(request, case_id, mobile_number, fulfilment_code_array)
```

Comment:
can't we do away with `fulfilment_code_array` and just use `available_fulfilments` here ?